lib/variable.js

```diff
@@ -32,15 +45,23 @@ module.exports = function getVariableAccessors (container, varName, defValue, ex
       let value = container[varName]
 
       if (typeof value === 'undefined') {
-        if (typeof defValue === 'undefined') {
-          // Need to return since no value is available. If a value needed to
-          // be available required() should be called, or a default passed
-          return
+        if (typeof defValue === 'undefined' && isRequired) {
+          // Var is not set, nor is a default. Throw an error
+          raiseError(undefined, 'is a required variable, but it was not set')
+        } else if (defValue) {
+          value = defValue
+        } else {
+          // return undefined since variable is not required and
+          // there's no default value provided
+          return undefined
         }
+      }
 
-        // Assign the default as the value since process.env does not contain
-        // the desired variable
-        value = defValue
+      if (isRequired) {
+        // Need to verify that required variables aren't just whitespace
+        if (value.trim().length === 0) {
```

Comment:
❕ This appears to be new functionality that might break existing applications — is there a way that we can make this behaviour optional?

E.g., `env.get(...).trim().required()`?